frontend/src/components/comp-top-table-bar/comp-top-table-bar.tsx

```diff
@@ -0,0 +1,106 @@
+import { useEffect, useRef, useState } from 'react';
+
+import { ButtonOutline } from '@components/common/button-outline/button-outline';
+import { TrashCanIcon } from '@components/common/icons/trash-can/trash-can';
+import { Spinner } from '@components/common/spinner/spinner';
+import {
+  useClearComparisonMutation,
+  useGetComparisonCarsQuery,
+} from '@store/queries/comparisons';
+import { clsx } from 'clsx';
+
+import { Comparison } from './components/comparison';
+import { NoActiveComparison } from './components/no-active-comparison';
+import styles from './components/styles.module.scss';
+
+export const CompTopTableBar = (): React.ReactElement => {
+  const { data, isLoading } = useGetComparisonCarsQuery();
+  const [clearTable] = useClearComparisonMutation();
+  const [isCleared, setIsCleared] = useState(false);
+  const ref = useRef<HTMLInputElement>();
+
+  const cars = data?.slice().sort((a, b) => a.position - b.position);
+  const ids = cars?.map((car) => car.id);
```

Comment:
`ids` name is too generic.

`.slice` and `.map` always generate new arrays! New arrays will be created on each rerender. Wrap then in `useMemo` if needed